### ISSUE TITLE
`RecoveryLogger` should accept interface instead of struct!

### ISSUE THREAD

**Issue by pavelnikolov** [gorilla/handlers#92]
This code is not designed well: https://github.com/gorilla/handlers/blob/master/recovery.go#L49

```
func RecoveryLogger(logger *log.Logger) RecoveryOption {
	return func(h http.Handler) {
		r := h.(*recoveryHandler)
		r.logger = logger
	}
}
```

You should avoid accepting structs and instead try to accept interfaces whenever possible. In this particular case, since you are only using the `Println(...interface{})` method, you should create an interface for the input argument like this:

```
type Logger interface {
    Println(v ...interface{}}
}
```

This would allow users to pass a custom structured logger (e.g. Logrus), which might have some other fields/tags like environment, host, user, application etc. Unstructured logs in a highly distributed environment are totally useless.

**Comment by kisielk** [comment]
I agree it probably should have used an interface. Not sure if changing that now would break existing code or not. I suspect it would be fine...

**Comment by pavelnikolov** [comment]
It doesn't break anything. I'll submit a PR soon.

**Comment by kisielk** [comment]
Fixed by #94
